src/Minesweeper.js

```diff
@@ -0,0 +1,131 @@
+import React, { Component } from 'react';
+import reactDom from 'react-dom';
+import PropTypes from 'prop-types';
+import Board from './components/Board/Board';
+import BoardHead from './components/BoardHead/BoardHead';
+
+ class Minesweeper extends React.Component {
+
+    constructor(props){
+        super(props);
+     
+        this.state = {
+                rows: props.rows,
+                columns : props.columns,
+                flags: 0,
+                mines: props.mines,
+                time: 0,
+                status: 0, //at game starts: waiting, running, ended
+                openCells : 0
+
+
+
+        };
+        // this.openGameClick = this.openGameClick.bind(this);
+        this.increamentFlags = this.increamentFlags.bind(this);
+        this.increamentOpenCells = this.increamentOpenCells.bind(this);
+        this.updateStatus = this.updateStatus.bind(this);
+        this.startTimer = this.startTimer.bind(this);
+        this.tick = this.tick.bind(this);
+    
+    }
+
+
+
+    // a function that increse the time every second
+    tick = ()=>
+    {
+        if (this.state.status == 1) {
+            let newtime = this.state.time + 1;
+            this.setState({time:newtime});
```

Comment:
When the new state depends on the old one, you do it that way:
```javascript
// Correct
this.setState((state, props) => ({
  counter: state.counter + props.increment
}));
```
read more at: https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous